koku/api/iam/view/user_preference.py

```diff
@@ -0,0 +1,272 @@
+#
+# Copyright 2018 Red Hat, Inc.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <https://www.gnu.org/licenses/>.
+#
+
+"""View for User Preferences."""
+
+from rest_framework import exceptions, mixins, viewsets
+from rest_framework.authentication import (SessionAuthentication,
+                                           TokenAuthentication)
+from rest_framework.permissions import IsAuthenticated
+
+import api.iam.models as models
+import api.iam.serializers as serializers
+from api.common.permissions.owner import IsOwner
+
+
+class UserPreferenceViewSet(mixins.CreateModelMixin,
+                            mixins.DestroyModelMixin,
+                            mixins.ListModelMixin,
+                            mixins.RetrieveModelMixin,
+                            mixins.UpdateModelMixin,
+                            viewsets.GenericViewSet):
+    """User Preference View.
+
+    A viewset that provides default `create()`, `destroy`, `retrieve()`,
+    and `list()` actions.
+    """
+
+    lookup_field = 'uuid'
+    queryset = models.UserPreference.objects.all()
+    serializer_class = serializers.UserPreferenceSerializer
+    authentication_classes = (TokenAuthentication,
+                              SessionAuthentication)
+    permission_classes = (IsOwner, IsAuthenticated)
+
+    def get_queryset(self):
+        """Get a queryset that only displays the owner's preferences."""
+        return self.queryset.filter(user=self.request.user)
```

Comment:
With this queryset in place do we really need the permission `IsOwner`, since they can only see and act upon preferences that are associated with the calling user?